apps/api/lib/api/web/controllers/user_controller.ex

```diff
@@ -1,6 +1,28 @@
 defmodule API.Web.UserController do
   use API.Web, :controller
 
+  def search(conn, %{"email" => email}) do
+    users = API.User.search_users(email: email)
+
+    conn
+    |> put_status(:ok)
+    |> render(%{users: users})
+  end
+
+  def search(conn, %{"username" => username}) do
+    users = API.User.search_users(username: username)
+    conn
+    |> put_status(:ok)
+    |> render(%{users: users})
+  end
+
+  def search(conn, _params) do
```

Comment:
Instead of handling edge cases here, we can use the action fallback controller -- https://hexdocs.pm/phoenix/Phoenix.Controller.html#action_fallback/1